pkg/templates/dnscache/deployment.go

```diff
@@ -0,0 +1,272 @@
+/*
+Copyright 2019 The KubeOne Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package dnscache
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+
+	"github.com/kubermatic/kubeone/pkg/util"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+const (
+	imagetag         = "k8s.gcr.io/k8s-dns-node-cache:1.15.2"
+	dnscacheCorefile = `
+cluster.local:53 {
```

Comment:
What if services DNS name is not `cluster.local`? Is it relevant and still going to work as expected?

Reply:
fixed